docs/src/pages/components/data-grid/components/CustomColumnMenu.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import {
+  GridColumnMenu,
+  GridColumnMenuContainer,
+  GridColumnMenuProps,
+  GridFilterMenuItem,
+  SortGridMenuItems,
+  useGridApiRef,
+  XGrid,
+} from '@material-ui/x-grid';
+import StarOutlineIcon from '@material-ui/icons/StarOutline';
+
+const useStyles = makeStyles((theme) => ({
+  primary: {
+    background: theme.palette.primary.main,
+    color: theme.palette.primary.contrastText,
+  },
+  secondary: {
+    background: theme.palette.secondary.main,
+    color: theme.palette.secondary.contrastText,
+  },
+}));
+
+export function CustomColumnMenuComponent(
+  props: GridColumnMenuProps & { color: string },
+) {
+  const classes = useStyles();
+  const { hideMenu, currentColumn, color, ...other } = props;
+
+  if (currentColumn.field === 'name') {
+    return (
+      <GridColumnMenuContainer
+        hideMenu={hideMenu}
+        currentColumn={currentColumn}
+        className={classes[color]}
+        {...other}
+      >
+        <SortGridMenuItems onClick={hideMenu} column={currentColumn!} />
+        <GridFilterMenuItem onClick={hideMenu} column={currentColumn!} />
+      </GridColumnMenuContainer>
+    );
+  }
+  if (currentColumn.field === 'stars') {
+    return (
+      <GridColumnMenuContainer
+        hideMenu={hideMenu}
+        currentColumn={currentColumn}
+        className={classes[color]}
+        {...other}
+      >
+        <div
+          style={{
+            width: 127,
+            height: 160,
+            display: 'flex',
+            justifyContent: 'center',
+            flexDirection: 'column',
+            alignItems: 'center',
+          }}
+        >
+          <StarOutlineIcon style={{ fontSize: 80 }} />
+        </div>
+      </GridColumnMenuContainer>
+    );
+  }
+  return (
+    <GridColumnMenu
+      hideMenu={hideMenu}
+      currentColumn={currentColumn}
+      className={classes[color]}
+      {...other}
+    />
+  );
+}
+
+export default function CustomColumnMenu() {
+  const [color, setColor] = React.useState<'primary' | 'secondary'>('primary');
+  const apiRef = useGridApiRef();
+
+  React.useEffect(() => {
+    apiRef.current.showColumnMenu('default');
+  }, [apiRef, color]);
+
+  return (
+    <div
+      style={{
+        height: 300,
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+      }}
+    >
+      <Button
```

Comment:
The UX feels weird full width. AFAIK, full width would only works on mobile for smaller screen widths

<img width="877" alt="Screenshot 2021-05-13 at 18 43 52" src="https://user-images.githubusercontent.com/3165635/118157708-31f21f00-b41b-11eb-9820-7736bff40ff5.png">